<h1> Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3> Analyst : Jomel Tomeo </h3>

In [1]:
# Libraries

import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3> Converting Dataset(s) into Dataframe </h3>
<h3> Requirements </h3>
<ul>
    <li> Name of the Dataframe </li>
    <li> File Path </li>
</ul>

In [2]:
penguins = pd.read_csv("datasets\\penguins.csv")
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h3> Descriptive Analytics </h3>
<hr>

In [3]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

In [4]:
penguins.describe().T

count         mean         std     min       25%      50%  \
rowid              344.0   172.500000   99.448479     1.0    86.750   172.50   
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                       75%     max  
rowid               258.25   344.0  
bill_length_mm       48.50    59.6  
bill_depth_mm        18.70    21.5  
flipper_length_mm   213.00   231.0  
body_mass_g        4750.00  6300.0  
year               2009.00  2009.0

<h3> Checking Null Values </h3>

In [5]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3> Filtering Column </h3>

In [6]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [7]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [8]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [9]:
penguins[penguins['species'] == 'Chinstrap']

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [10]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [11]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

In [12]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [13]:
chinstrap_heavy = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]
chinstrap_heavy

rowid    species island  bill_length_mm  bill_depth_mm  \
284    285  Chinstrap  Dream            46.0           18.9   
289    290  Chinstrap  Dream            52.0           18.1   
291    292  Chinstrap  Dream            50.5           19.6   
295    296  Chinstrap  Dream            49.2           18.2   
301    302  Chinstrap  Dream            52.0           19.0   
305    306  Chinstrap  Dream            52.8           20.0   
307    308  Chinstrap  Dream            54.2           20.8   
309    310  Chinstrap  Dream            51.0           18.8   
313    314  Chinstrap  Dream            52.0           20.7   
315    316  Chinstrap  Dream            53.5           19.9   
321    322  Chinstrap  Dream            50.8           18.5   
323    324  Chinstrap  Dream            49.0           19.6   
329    330  Chinstrap  Dream            50.7           19.7   
333    334  Chinstrap  Dream            49.3           19.9   
339    340  Chinstrap  Dream            55.8           19.8   
342    343  Chinstrap  Dream            50.8           19.0   

     flipper_length_mm  body_mass_g     sex  year  
284              195.0       4150.0  female  2007  
289              201.0       4050.0    male  2007  
291              201.0       4050.0    male  2007  
295              195.0       4400.0    male  2007  
301              197.0       4150.0    male  2007  
305              205.0       4550.0    male  2008  
307              201.0       4300.0    male  2008  
309              203.0       4100.0    male  2008  
313              210.0       4800.0    male  2008  
315              205.0       4500.0    male  2008  
321              201.0       4450.0    male  2009  
323              212.0       4300.0    male  2009  
329              203.0       4050.0    male  2009  
333              203.0       4050.0    male  2009  
339              207.0       4000.0    male  2009  
342              210.0       4100.0    male  2009

In [14]:
chinstrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              16 non-null     int64  
 1   species            16 non-null     object 
 2   island             16 non-null     object 
 3   bill_length_mm     16 non-null     float64
 4   bill_depth_mm      16 non-null     float64
 5   flipper_length_mm  16 non-null     float64
 6   body_mass_g        16 non-null     float64
 7   sex                16 non-null     object 
 8   year               16 non-null     int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 1.2+ KB


<h3>Filling-Up Null Values</h3>

In [15]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [16]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [17]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Changing the Data Type</h2>

In [18]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [19]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
dtypes: category(4), float64(4), int64(1)
memory usage: 15.4 KB


<h3>Exploring Data Analytics</h3>
<hr>

<h3>Sample 1</h1>
<h4>Which specie has the heaviest body mass?</h4>

In [20]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df['body_mass_g'].mean()

np.float64(3703.95891043398)

In [21]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df['body_mass_g'].mean()

np.float64(5068.965761177136)

In [22]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

<h4>Sample 1 Insights</h4>
<p> Gentoo has the heaviest weight recorded with 5076 body mass.</p>

<h2> How to create a column</h2>

In [23]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
dtypes: category(4), float64(4), int64(1)
memory usage: 15.4 KB


In [24]:
penguins.columns

Index(['rowid', 'species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [25]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 1)
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
0           181.000000  3750.000000    male  2007      3.8  
1           186.000000  3800.000000  female  2007      3.8  
2           195.000000  3250.000000  female  2007      3.2  
3           200.915205  4201.754386    male  2007      4.2  
4           193.000000  3450.000000  female  2007      3.4  
..                 ...          ...     ...   ...      ...  
339         207.000000  4000.000000    male  2009      4.0  
340         202.000000  3400.000000  female  2009      3.4  
341         193.000000  3775.000000    male  2009      3.8  
342         210.000000  4100.000000    male  2009      4.1  
343         198.000000  3775.000000  female  2009      3.8  

[344 rows x 10 columns]

In [26]:
penguins['mass_kg'].max()

np.float64(6.3)

In [27]:
penguins[penguins['mass_kg'] == 6.3 ]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169    170  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  mass_kg  
169       6300.0  male  2007      6.3

In [28]:
penguins['bill_length'] = np.round(penguins['bill_length_mm'] / 100, 1)
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  bill_length  
0           181.000000  3750.000000    male  2007      3.8          0.4  
1           186.000000  3800.000000  female  2007      3.8          0.4  
2           195.000000  3250.000000  female  2007      3.2          0.4  
3           200.915205  4201.754386    male  2007      4.2          0.4  
4           193.000000  3450.000000  female  2007      3.4          0.4  
..                 ...          ...     ...   ...      ...          ...  
339         207.000000  4000.000000    male  2009      4.0          0.6  
340         202.000000  3400.000000  female  2009      3.4          0.4  
341         193.000000  3775.000000    male  2009      3.8          0.5  
342         210.000000  4100.000000    male  2009      4.1          0.5  
343         198.000000  3775.000000  female  2009      3.8          0.5  

[344 rows x 11 columns]

In [29]:
filtered_islands = penguins[penguins['mass_kg'] >= 5.4]['island']
filtered_islands

153    Biscoe
155    Biscoe
156    Biscoe
163    Biscoe
165    Biscoe
167    Biscoe
169    Biscoe
171    Biscoe
172    Biscoe
179    Biscoe
181    Biscoe
185    Biscoe
187    Biscoe
191    Biscoe
193    Biscoe
196    Biscoe
199    Biscoe
211    Biscoe
215    Biscoe
217    Biscoe
219    Biscoe
221    Biscoe
227    Biscoe
229    Biscoe
231    Biscoe
233    Biscoe
235    Biscoe
237    Biscoe
241    Biscoe
243    Biscoe
245    Biscoe
253    Biscoe
255    Biscoe
257    Biscoe
259    Biscoe
261    Biscoe
263    Biscoe
265    Biscoe
267    Biscoe
269    Biscoe
273    Biscoe
275    Biscoe
Name: island, dtype: category
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [30]:
median_flipper_chinstrap = penguins[penguins['species'] == 'Chinstrap']['flipper_length_mm']
median_flipper_chinstrap

276    192.0
277    196.0
278    193.0
279    188.0
280    197.0
       ...  
339    207.0
340    202.0
341    193.0
342    210.0
343    198.0
Name: flipper_length_mm, Length: 68, dtype: float64

<h2> CORRELATIONAL ANALYSIS</h2>
<hr>
<h3> Testing Correlation using Pearson R</h3>
<hr>
<p><strong>Pearson R</strong> can test correlation between two (2) variables, given that both variables are <strong>NUMERICAL DATA</strong></p>


In [31]:
flipper_mass = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])
flipper_mass

np.float64(0.8712017673060115)

<strong>CATEGORICAL DATA</strong>
<hr>

In [32]:
# Import scipy.stats
from scipy.stats import f_oneway

In [33]:
# Filter male penguins body mass
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [34]:
# Filter female penguins body mass
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [35]:
stats, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'], penguins[penguins['sex'] == 'female']['body_mass_g'])
p_value

np.float64(3.0226600487050246e-15)

<h1>DATA ANALYTICS</h1>
<hr>
<h2> Driving data exploration by formulating and analyzing thirteen (13) targeted questions </h2>
<hr>
<h4> Listed below are twenty-five (25) questions that shape the analytical process</h4>
<ol>
    <li><p> Identify the typical flipper length observed among Chinstrap penguins. </p></li>
    <li><p> Identify all penguins that have either a bill length greater than 55mm or a bill depth greater than 20mm or both </p></li>
    <li><p> Determine the count of male individuals in the Adelie penguin (Pygoscelis adeliae) population sample. </p></li>
    <li><p> Which islands do penguins live on, and how many live on each island? </p></li>
    <li><p> Do penguins with longer flippers tend to be heavier? </p></li>
    <li><p> What is the total number of female penguins recorded in the year 2008? </p></li>
    <li><p> What are the details (island, sex, body mass, etc.) of the Gentoo penguin with the maximum bill length? </p></li>
    <li><p> What is the total body mass of all Gentoo penguins by sex? </p></li>
    <li><p> What percentage of each species' population is male?</p></li>
    <li><p> In what direction are bill length and bill depth related among Gentoo penguins? </p></li>
    <li><p> Determine the central flipper length measurement for male and female Chinstrap penguins on Dream Island. </p></li>
    <li><p> Show the distribution of penguin species for each study year. </p></li>
    <li><p>Does body mass relate more strongly to flipper length or to bill length in penguins? </p></li>
</ol>
<hr>

<h2> Question 1 </h2>
<h3> Identify the typical flipper length observed among Chinstrap penguins. </h3>

In [36]:
median_flipper_chinstrap = penguins[penguins['species'] == 'Chinstrap']['flipper_length_mm'].median()
median_flipper_chinstrap

np.float64(196.0)

<span style="color:red;"><b>INSIGHTS:</b></span>  The typical flipper length observed among Chinstrap penguins is 196.0 mm. This represents the middle point of the flipper length measurements, suggesting that half of the observed Chinstrap penguins have flipper lengths shorter than 196.0 mm, and the other half have lengths longer than this value.
<hr>


<h2> Question 2 </h2>
<h3> Identify all penguins that have either a bill length greater than 55mm or a bill depth greater than 20mm or both </h3>

In [37]:
penguins[(penguins['bill_length_mm'] > 55) | (penguins['bill_depth_mm'] > 20)]

rowid    species     island  bill_length_mm  bill_depth_mm  \
5        6     Adelie  Torgersen            39.3           20.6   
9       10     Adelie  Torgersen            42.0           20.2   
13      14     Adelie  Torgersen            38.6           21.2   
14      15     Adelie  Torgersen            34.6           21.1   
17      18     Adelie  Torgersen            42.5           20.7   
19      20     Adelie  Torgersen            46.0           21.5   
35      36     Adelie      Dream            39.2           21.1   
49      50     Adelie      Dream            42.3           21.2   
61      62     Adelie     Biscoe            41.3           21.1   
85      86     Adelie      Dream            41.3           20.3   
111    112     Adelie     Biscoe            45.6           20.3   
114    115     Adelie     Biscoe            39.6           20.7   
117    118     Adelie  Torgersen            37.3           20.5   
137    138     Adelie      Dream            40.2           20.1   
185    186     Gentoo     Biscoe            59.6           17.0   
253    254     Gentoo     Biscoe            55.9           17.0   
267    268     Gentoo     Biscoe            55.1           16.0   
287    288  Chinstrap      Dream            51.7           20.3   
293    294  Chinstrap      Dream            58.0           17.8   
307    308  Chinstrap      Dream            54.2           20.8   
313    314  Chinstrap      Dream            52.0           20.7   
339    340  Chinstrap      Dream            55.8           19.8   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  bill_length  
5                190.0       3650.0    male  2007      3.6          0.4  
9                190.0       4250.0    male  2007      4.2          0.4  
13               191.0       3800.0    male  2007      3.8          0.4  
14               198.0       4400.0    male  2007      4.4          0.3  
17               197.0       4500.0    male  2007      4.5          0.4  
19               194.0       4200.0    male  2007      4.2          0.5  
35               196.0       4150.0    male  2007      4.2          0.4  
49               191.0       4150.0    male  2007      4.2          0.4  
61               195.0       4400.0    male  2008      4.4          0.4  
85               194.0       3550.0    male  2008      3.6          0.4  
111              191.0       4600.0    male  2009      4.6          0.5  
114              191.0       3900.0  female  2009      3.9          0.4  
117              199.0       3775.0    male  2009      3.8          0.4  
137              200.0       3975.0    male  2009      4.0          0.4  
185              230.0       6050.0    male  2007      6.0          0.6  
253              228.0       5600.0    male  2009      5.6          0.6  
267              230.0       5850.0    male  2009      5.8          0.6  
287              194.0       3775.0    male  2007      3.8          0.5  
293              181.0       3700.0  female  2007      3.7          0.6  
307              201.0       4300.0    male  2008      4.3          0.5  
313              210.0       4800.0    male  2008      4.8          0.5  
339              207.0       4000.0    male  2009      4.0          0.6

In [38]:
big_bill_penguins = penguins[(penguins['bill_length_mm'] > 55) | (penguins['bill_depth_mm'] > 20)]
big_bill_penguins['species'].value_counts()

species
Adelie       14
Chinstrap     5
Gentoo        3
Name: count, dtype: int64

<span style="color:red;"><b>INSIGHTS:</b></span> The study identified 22 penguins meeting the criteria of either having bill lengths exceeding 55 mm or bill depths greater than 20 mm. 

Adelie penguins dominated this group (14 individuals), nearly all characterized by notably thick bills over 20 mm, though their lengths remained under 55 mm. In contrast, the three (3 individuals)  Gentoo penguins exhibited elongated bills surpassing 55 mm, but with more moderate depths below 20 mm. Chinstrap penguins (5 individuals) displayed intermediate characteristics, with some specimens showing both extended lengths and substantial depths, while others featured just one of these traits.
<hr>

<h2> Question 3 </h2>
<h3> Determine the count of male individuals in the Adelie penguin (Pygoscelis adeliae) population sample. </h3>

In [39]:
(penguins['species'] == 'Adelie') & (penguins['sex'] == 'male')

0       True
1      False
2      False
3       True
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [40]:
penguins[(penguins['species'] == 'Adelie')]['sex'].value_counts()['male']

np.int64(79)

<span style="color:red;"><b>INSIGHTS:</b></span> A total of 79 male individuals were identified within the Adelie penguin (Pygoscelis adeliae) population sample.
<hr>

<h2> Question 4 </h2>
<h3> Which islands do penguins live on, and how many live on each island? </h3>

In [41]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<span style="color:red;"><b>INSIGHTS:</b></span> Penguins primarily inhabit three islands: Biscoe, Dream, and Torgersen.

Biscoe Island supports the largest colony with 168 penguins (48.8% of total population), followed by Dream Island with 124 individuals (36.0%), and Torgersen Island with the smallest group of 52 penguins (15.2%).

These findings clearly establish Biscoe as the preferred habitat, hosting nearly half of all observed penguins, while Torgersen represents the least populated location.
<hr>


<h2> Question 5 </h2>
<h3> Do penguins with longer flippers (over 200mm) tend to be heavier? </h3>

In [42]:
penguins[penguins['flipper_length_mm'] > 200]['body_mass_g'].mean()

np.float64(4905.690058479533)

In [43]:
penguins[penguins['flipper_length_mm'] < 200]['body_mass_g'].mean()

np.float64(3654.342105263158)

<span style="color:red;"><b>INSIGHTS:</b></span> Penguins with longer flippers (exceeding 200 mm) have a mean body mass of 4,905.69 grams, whereas those with shorter flippers (under 200 mm) average 3,654.34 grams. There's a substantial difference of 1,251.35 grams.

In that case, penguins with longer flippers (>200 mm) are significantly heavier, indicating a 34% higher mean body mass compared to their shorter-flippered counterparts.
<hr>

<h2> Question 6 </h2>
<h3> What is the total number of female penguins recorded in the year 2008? </h3>

In [44]:
penguins[(penguins['year'] == 2008) & (penguins['sex'] == 'female')]

rowid    species     island  bill_length_mm  bill_depth_mm  \
50      51     Adelie     Biscoe            39.6           17.7   
52      53     Adelie     Biscoe            35.0           17.9   
54      55     Adelie     Biscoe            34.5           18.1   
56      57     Adelie     Biscoe            39.0           17.5   
58      59     Adelie     Biscoe            36.5           16.6   
60      61     Adelie     Biscoe            35.7           16.9   
62      63     Adelie     Biscoe            37.6           17.0   
64      65     Adelie     Biscoe            36.4           17.1   
66      67     Adelie     Biscoe            35.5           16.2   
68      69     Adelie  Torgersen            35.9           16.6   
70      71     Adelie  Torgersen            33.5           19.0   
72      73     Adelie  Torgersen            39.6           17.2   
74      75     Adelie  Torgersen            35.5           17.5   
76      77     Adelie  Torgersen            40.9           16.8   
78      79     Adelie  Torgersen            36.2           16.1   
80      81     Adelie  Torgersen            34.6           17.2   
82      83     Adelie  Torgersen            36.7           18.8   
84      85     Adelie      Dream            37.3           17.8   
87      88     Adelie      Dream            36.9           18.6   
89      90     Adelie      Dream            38.9           18.8   
90      91     Adelie      Dream            35.7           18.0   
92      93     Adelie      Dream            34.0           17.1   
94      95     Adelie      Dream            36.2           17.3   
96      97     Adelie      Dream            38.1           18.6   
98      99     Adelie      Dream            33.1           16.1   
186    187     Gentoo     Biscoe            49.1           14.8   
188    189     Gentoo     Biscoe            42.6           13.7   
190    191     Gentoo     Biscoe            44.0           13.6   
192    193     Gentoo     Biscoe            42.7           13.7   
194    195     Gentoo     Biscoe            45.3           13.7   
197    198     Gentoo     Biscoe            43.6           13.9   
198    199     Gentoo     Biscoe            45.5           13.9   
200    201     Gentoo     Biscoe            44.9           13.3   
202    203     Gentoo     Biscoe            46.6           14.2   
204    205     Gentoo     Biscoe            45.1           14.4   
206    207     Gentoo     Biscoe            46.5           14.4   
208    209     Gentoo     Biscoe            43.8           13.9   
210    211     Gentoo     Biscoe            43.2           14.5   
212    213     Gentoo     Biscoe            45.3           13.8   
214    215     Gentoo     Biscoe            45.7           13.9   
216    217     Gentoo     Biscoe            45.8           14.2   
220    221     Gentoo     Biscoe            43.5           14.2   
222    223     Gentoo     Biscoe            47.7           15.0   
225    226     Gentoo     Biscoe            46.5           14.8   
226    227     Gentoo     Biscoe            46.4           15.0   
228    229     Gentoo     Biscoe            47.5           14.2   
230    231     Gentoo     Biscoe            45.2           13.8   
302    303  Chinstrap      Dream            50.5           18.4   
304    305  Chinstrap      Dream            46.4           17.8   
306    307  Chinstrap      Dream            40.9           16.6   
308    309  Chinstrap      Dream            42.5           16.7   
311    312  Chinstrap      Dream            47.5           16.8   
312    313  Chinstrap      Dream            47.6           18.3   
314    315  Chinstrap      Dream            46.9           16.6   
317    318  Chinstrap      Dream            46.2           17.5   
319    320  Chinstrap      Dream            45.5           17.0   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  bill_length  
50               186.0       3500.0  female  2008      3.5          0.4  
52               190.0       3450.0  

In [45]:
female_2008_count = ((penguins['year'] == 2008) & (penguins['sex'] == 'female')).sum()
female_2008_count

np.int64(56)

<span style="color:red;"><b>INSIGHTS:</b></span> Throughout the year 2008, a total of 56 female penguins were recorded and included in the annual database of penguin demographics.
<hr>

<h2> Question 7 </h2>
<h3> What are the details (island, sex, body mass, etc.) of the Gentoo penguin with the maximum bill length? </h3>

In [46]:
gentoo = penguins[penguins['species'] == 'Gentoo']

In [47]:
gentoo_longest_bill = gentoo[gentoo['bill_length_mm'] == gentoo['bill_length_mm'].max()]
gentoo_longest_bill

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
185    186  Gentoo  Biscoe            59.6           17.0              230.0   

     body_mass_g   sex  year  mass_kg  bill_length  
185       6050.0  male  2007      6.0          0.6

<span style="color:red;"><b>INSIGHTS:</b></span> The Gentoo penguin with the longest bill, measuring 59.6 mm, was a large male weighing 6,050 g. He was from Biscoe Island, had 230 mm flippers, and was recorded in 2007.
<hr>

<h2> Question 8 </h2>
<h3> What is the total body mass of all Gentoo penguins by sex? </h3>

In [48]:
penguins[penguins['species'] == 'Gentoo'].groupby('sex')['body_mass_g'].sum()

sex
female    271425.000000
male      357126.754386
Name: body_mass_g, dtype: float64

<span style="color:red;"><b>INSIGHTS:</b></span> There is a difference in total body mass between male and female Gentoo penguins. The combined body mass of all recorded males is approximately 31.6% higher than that of females (357,127g vs. 271,425g). Males are typically larger and heavier than females. 

This size difference can be attributed to roles in breeding and competition, where larger body mass provides males with an advantage in securing mates and defending territories.
<hr>

<h2> Question 9 </h2>
<h3> What percentage of each species' population is male? </h3>

In [49]:
species_count = penguins['species'].value_counts()
male_count = penguins[penguins['sex'] == 'male'].groupby('species').size()
male_percentage = (male_count / species_count * 100).round(2)
male_percentage

species
Adelie       51.97
Chinstrap    50.00
Gentoo       53.23
dtype: float64

<span style="color:red;"><b>INSIGHTS:</b></span> 

The male percentage is nearly a perfect 50/50 sex ratio across all three penguin species, with Adelie, Chinstrap, and Gentoo populations being 51.97%, 50.00%, and 53.23% male, respectively.

The numbers tell us the sex ratio is perfectly balanced across all species. This is a sign of a healthy, stable population. The very small differences between species are normal and not a cause for concern. In nature, a 50/50 split is common and suggests there are no major environmental pressures favoring one sex over the other.
<hr>

<h2> Question 10 </h2>
<h3> In what direction are bill length and bill depth related among Gentoo penguins? </h3>

In [50]:
gentoo_corr = penguins[penguins['species'] == 'Gentoo']['bill_length_mm'].corr(penguins[penguins['species'] == 'Gentoo']['bill_depth_mm']).round(3)
print(f"Gentoo Bill Length/Depth Correlation: {gentoo_corr}")

Gentoo Bill Length/Depth Correlation: 0.607


<span style="color:red;"><b>INSIGHTS:</b></span> For Gentoo penguins, bill length and bill depth are positively correlated with a coefficient of 0.607. This is a strong relationship, meaning Gentoo penguins with longer bills also tend to have deeper bills. While the association is not perfect, it is consistent enough that bill length can be considered a fairly good predictor of bill depth within this species.
<hr>

<h2> Question 11 </h2>
<h3> Determine the central flipper length measurement for male and female Chinstrap penguins on Dream Island. </h3>

In [51]:
chinstrap_flipper = penguins[(penguins['species'] == 'Chinstrap') & (penguins['island'] == 'Dream')].groupby('sex')['flipper_length_mm'].median()
chinstrap_flipper

sex
female    192.0
male      200.5
Name: flipper_length_mm, dtype: float64

<span style="color:red;"><b>INSIGHTS:</b></span> This illustrates the difference between male and female Chinstrap penguins on Dream Island. The median flipper length for females is 192.0 mm, while for males it is 200.5 mm. 

This 8.5 mm difference shows that male Chinstrap penguins are consistently larger than females, a common biological pattern known as sexual dimorphism.
<hr>

<h2> Question 12 </h2>
<h3> Show the distribution of penguin species for each study year. </h3>

In [52]:
year_species = penguins.groupby(['year', 'species']).size()
print(year_species)

year  species  
2007  Adelie       50
      Chinstrap    26
      Gentoo       34
2008  Adelie       50
      Chinstrap    18
      Gentoo       46
2009  Adelie       52
      Chinstrap    24
      Gentoo       44
dtype: int64


<span style="color:red;"><b>INSIGHTS:</b></span> This shows the distribution of penguin species across the three-year study period from 2007 to 2009.

The Adelie population remained very stable each year, with around 50 individuals. The Gentoo population displayed a positive trend, increasing from 34 to 44 individuals. In contrast, the Chinstrap population was more variable, dropping to 18 individuals in 2008 before recovering in 2009.

This may mean that changes in the environment or in how the penguins were observed affected the Chinstrap population during the study period.
<hr>

<h2> Question 13 </h2>
<h3> Does body mass relate more strongly to flipper length or to bill length in penguins? </h3>

In [53]:
corr_mass_flipper = penguins['body_mass_g'].corr(penguins['flipper_length_mm']).round(3)
corr_mass_bill = penguins['body_mass_g'].corr(penguins['bill_length_mm']).round(3)
print(f"Correlation (Mass vs. Flipper): {corr_mass_flipper}")
print(f"Correlation (Mass vs. Bill Length): {corr_mass_bill}")

Correlation (Mass vs. Flipper): 0.871
Correlation (Mass vs. Bill Length): 0.595


<span style="color:red;"><b>INSIGHTS:</b></span> The correlation between body mass and flipper length is 0.871, indicating a very strong positive relationship. This means that as penguins’ body mass increases, their flipper length also tends to increase in a highly consistent way. In contrast, the correlation between body mass and bill length is 0.595, which represents a moderate positive relationship. Heavier penguins generally have longer bills, but the connection is not as strong.
<hr>

<div style="font-family: Arial, sans-serif; font-size: 14px; color: #222; line-height: 1.6; margin-top: 40px; text-align: center;">
  <div style="margin-bottom: 15px;">
    <p style="margin: 5px 0; text-align: center;"><strong>Data Analysis by:</strong> Jomel Tomeo</p>
    <p style="margin: 5px 0; text-align: center;"><strong>Institution/Organization:</strong> Dalubhasaan ng Lungsod ng Lucena</p>
    <p style="margin: 5px 0; text-align: center;"><strong>Date:</strong> August 08, 2025</p>
  </div>

  <div style="width: 50%; margin: 20px auto; border-top: 1px solid #aaa;"></div>

  <div style="font-size: 12px; color: #666; margin-top: 20px; max-width: 600px; margin-left: auto; margin-right: auto; text-align: center;">
    © 2025 All rights reserved. This analysis is prepared exclusively for academic and professional purposes. 
    No part of this work may be reproduced, distributed, or transmitted without prior permission.
  </div>
</div>
